From 27bc7f9d520a0f16d0b4a044bd9d56473c5e4cd6 Mon Sep 17 00:00:00 2001
From: Ravi B <ravibabu@ti.com>
Date: Sat, 30 Jun 2012 07:17:39 +0800
Subject: [PATCH 401/609] usb:musb: Fixes for code review of musb

Original commit: e15ee2863f229bcdb5cdb45228cdb7a68d4c9f97

  From git://arago-project.org/git/projects/linux-am33x.git
  And ti-sdk-am335x-evm-05.05.00.00-Linux-x86-Install image

Codes reviews for CPPI.

Signed-off-by: Ravi B <ravibabu@ti.com>
Signed-off-by: Vaibhav Hiremath <hvaibhav@ti.com>
---
 drivers/usb/gadget/omap_udc.c |    1 +
 drivers/usb/musb/am35x.c      |    4 +++-
 drivers/usb/musb/cppi41.h     |   10 ++++++++++
 drivers/usb/musb/musb_core.c  |    4 ++--
 drivers/usb/musb/ti81xx.c     |    9 +++++----
 5 files changed, 21 insertions(+), 7 deletions(-)

diff --git a/drivers/usb/gadget/omap_udc.c b/drivers/usb/gadget/omap_udc.c
index 3b4b6dd..e4fe171 100644
--- a/drivers/usb/gadget/omap_udc.c
+++ b/drivers/usb/gadget/omap_udc.c
@@ -2798,6 +2798,7 @@ static int __init omap_udc_probe(struct platform_device *pdev)
 	struct omap_usb_config	*config = pdev->dev.platform_data;
 	struct clk		*dc_clk;
 	struct clk		*hhc_clk;
+	u8			pdev_id;
 
 	/* NOTE:  "knows" the order of the resources! */
 	if (!request_mem_region(pdev->resource[0].start,
diff --git a/drivers/usb/musb/am35x.c b/drivers/usb/musb/am35x.c
index c7f68f7..b958b91 100644
--- a/drivers/usb/musb/am35x.c
+++ b/drivers/usb/musb/am35x.c
@@ -88,6 +88,8 @@
 #define USB_MENTOR_CORE_OFFSET	0x400
 
 #ifdef CONFIG_USB_TI_CPPI41_DMA
+#define CPPI41_QMGR_REG0SIZE	0x3fff
+
 /*
  * CPPI 4.1 resources used for USB OTG controller module:
  *
@@ -240,7 +242,7 @@ int __devinit cppi41_init(struct musb *musb)
 	cppi41_dma_block[0].sched_table_base = musb->ctrl_base + 0x2800;
 
 	/* Initialize for Linking RAM region 0 alone */
-	cppi41_queue_mgr_init(cppi_info->q_mgr, 0, 0x3fff);
+	cppi41_queue_mgr_init(cppi_info->q_mgr, 0, CPPI41_QMGR_REG0SIZE);
 
 	numch =  USB_CPPI41_NUM_CH * 2;
 	order = get_count_order(numch);
diff --git a/drivers/usb/musb/cppi41.h b/drivers/usb/musb/cppi41.h
index bc42ac2..9d42109 100644
--- a/drivers/usb/musb/cppi41.h
+++ b/drivers/usb/musb/cppi41.h
@@ -184,6 +184,16 @@
 
 #define CPPI41_TXDMA_MAXLEN		(4 * 1024 * 1024 - 1)
 #define CPPI41_RXDMA_MAXLEN		(64 * 1024)
+
+/*
+ * Queue Status register
+ */
+#define CPPI41_QSTATUS_REG0	0x90
+#define CPPI41_QSTATUS_REG1	0x94
+#define CPPI41_QSTATUS_REG2	0x98
+#define CPPI41_QSTATUS_REG3	0x9c
+#define CPPI41_QSTATUS_REG4	0xa0
+
 /*
  * DMA Scheduler - Table Region
  */
diff --git a/drivers/usb/musb/musb_core.c b/drivers/usb/musb/musb_core.c
index 9ba8dd5..11bab58 100644
--- a/drivers/usb/musb/musb_core.c
+++ b/drivers/usb/musb/musb_core.c
@@ -1292,9 +1292,9 @@ static int __devinit ep_config_from_table(struct musb *musb)
 	int			offset;
 	struct musb_hw_ep	*hw_ep = musb->endpoints;
 
-	if (musb->config->fifo_mode) {
+	if (musb->config->fifo_mode)
 		fifo_mode = musb->config->fifo_mode;
-	} else if (musb->config->fifo_cfg) {
+	else if (musb->config->fifo_cfg) {
 		cfg = musb->config->fifo_cfg;
 		n = musb->config->fifo_cfg_size;
 		goto done;
diff --git a/drivers/usb/musb/ti81xx.c b/drivers/usb/musb/ti81xx.c
index 3724950..6fb5a56 100644
--- a/drivers/usb/musb/ti81xx.c
+++ b/drivers/usb/musb/ti81xx.c
@@ -1,5 +1,5 @@
 /*
- * Texas Instruments TI81XX "glue layer"
+ * Texas Instruments TI81XX "usb platform glue layer"
  *
  * Copyright (c) 2008, MontaVista Software, Inc. <source@mvista.com>
  *
@@ -699,9 +699,10 @@ static irqreturn_t cppi41dma_Interrupt(int irq, void *hci)
 		usbss_write(USBSS_IRQ_STATUS, intr_status);
 	else
 		printk(KERN_DEBUG "spurious usbss intr\n");
-	q_cmpl_status_0 = musb_readl(q_mgr_base, 0x98);
-	q_cmpl_status_1 = musb_readl(q_mgr_base, 0x9c);
-	q_cmpl_status_2 = musb_readl(q_mgr_base, 0xa0);
+
+	q_cmpl_status_0 = musb_readl(q_mgr_base, CPPI41_QSTATUS_REG2);
+	q_cmpl_status_1 = musb_readl(q_mgr_base, CPPI41_QSTATUS_REG3);
+	q_cmpl_status_2 = musb_readl(q_mgr_base, CPPI41_QSTATUS_REG4);
 
 	/* USB0 tx/rx completion */
 	/* usb0 tx completion interrupt for ep1..15 */
-- 
1.7.5.4

